Code/GraphMol/Fingerprints/MHFP.h

```diff
@@ -0,0 +1,302 @@
+    \param isomeric whether the SMILES added to the shingling are isomeric.
+            Default: <tt>false</tt>.
+    \param kekulize whether the SMILES added to the shingling are kekulized.
+            Default: <tt>true</tt>.
+    \param min_radius the minimum radius that is used to extract n-grams.
+            Default: <tt>1</tt>.
+    \param length the length into which the fingerprint is folded.
+            Default: <tt>2048</tt>.
+
+    \returns the SECFP fingerprint.
+   */
+  ExplicitBitVect
+  EncodeSECFP(ROMol& mol,
+       const unsigned char& radius = 3,
+       const bool& rings = true,
+       const bool& isomeric = false,
+       const bool& kekulize = true,
+       const unsigned char& min_radius = 1,
+       const size_t& length = 2048);
+
+  //! \overload
+  std::vector<ExplicitBitVect>
+  EncodeSECFP(std::vector<ROMol>& mols,
+       const unsigned char& radius = 3,
+       const bool& rings = true,
+       const bool& isomeric = false,
+       const bool& kekulize = true,
+       const unsigned char& min_radius = 1,
+       const size_t& length = 2048);
+
+  //! \overload
+  ExplicitBitVect
+  EncodeSECFP(std::string& smiles,
+       const unsigned char& radius = 3,
+       const bool& rings = true,
+       const bool& isomeric = false,
+       const bool& kekulize = true,
+       const unsigned char& min_radius = 1,
+       const size_t& length = 2048);
+
+  //! \overload
+  std::vector<ExplicitBitVect>
+  EncodeSECFP(std::vector<std::string>& smiles,
+       const unsigned char& radius = 3,
+       const bool& rings = true,
+       const bool& isomeric = false,
+       const bool& kekulize = true,
+       const unsigned char& min_radius = 1,
+       const size_t& length = 2048);
+
+  /*!
+    \brief Calculates the Jaccard / Tanimoto distance between two MHFP fingerprints.
+   
+    \param a an MHFP fingerprint vector.
+    \param b an MHFP fingerprint vector.
+
+    \returns the Jaccard / Tanimoto distance between the two fingerprints.
+   */
+  static float
+  Distance(const std::vector<uint32_t>& a, 
+       const std::vector<uint32_t>& b);
+
+private:
+  //! The fastest mod implementation. 
+  uint64_t
+  FastMod(const uint64_t input, const uint64_t ceil) {
+    return input >= ceil ? input % ceil : input;
+  }
+
+  ExplicitBitVect
+  Fold(const std::vector<uint32_t>& vec, uint32_t length = 2048) {
+    ExplicitBitVect ebv(length);
+    for (size_t i = 0; i < vec.size(); i++)
+      ebv.setBit(vec[i] % length);
+    return ebv;
+  }
+
+  std::vector<uint32_t>
+  HashShingling(std::vector<std::string> vec) {
+    std::vector<uint32_t> result(vec.size());
+    for (size_t i = 0; i < vec.size(); i++)
+      result[i] = FNV::hash(vec[i]);
+    return result;
+  }
+
+  unsigned int n_permutations_, seed_;
+  uint64_t prime_ = 2305843009213693951UL;
+  uint32_t max_hash_ = 4294967295;
+  std::vector<uint32_t> perms_a_;
+  std::vector<uint32_t> perms_b_;
+  std::vector<std::vector<float>> rs_;
```

Comment:
Any particular reason to be using `float` here instead of `double`?

Reply:
The variables are actually not required.